### PR TITLE
[python] [docs] fixed objective in sklearn wrapper; added missed objectives & metrics to docs

### PR DESCRIPTION
**Pull request by StrikerRUS**
Refer to #1055.

### CODE REVIEW

File: python-package/lightgbm/sklearn.py [hunk 1]
```diff
@@ -285,6 +285,8 @@ def get_params(self, deep=True):
     def set_params(self, **params):
         for key, value in params.items():
             setattr(self, key, value)
+            if hasattr(self, '_' + key):
```
Comment (chivee) [inline python-package/lightgbm/sklearn.py]: if an params was named `other_params`, could that disable line 290?

**Comment by chivee** [comment]
@StrikerRUS , very detailed work, it will be great help for the users.

**Comment by StrikerRUS** [comment]
@guolinke I've added note about `ignore_column`

File: python-package/lightgbm/sklearn.py (hunk 1 above)
Comment (StrikerRUS) [reply python-package/lightgbm/sklearn.py]: @chivee Yeah, you're right!
Any parameter with underscore (`_Booster`, `_classes`, especially `_n_features`) will break normal logic in some way. I think that we're assuming that LightGBM user is rather smart person and wont do this 😃 .

**Comment by StrikerRUS** [comment]
@wxchan Please take a look at Python part of the code.

**Pull request merged**